src/core/labeling/qgstextlabelfeature.h

```diff
@@ -121,11 +110,19 @@ class CORE_EXPORT QgsTextLabelFeature : public QgsLabelFeature
     QgsTextDocument document() const;
 
     /**
-     * Sets the \a document for the label.
+     * Returns the document metrics for the label.
+     *
+     * \see document()
+     * \since QGIS 3.28
+     */
+    QgsTextDocumentMetrics documentMetrics() const;
+
+    /**
+     * Sets the \a document and document \a metrics for the label.
```

Comment:
Should we \note the metrics parameter appeared in QGIS 3.28? No strong feelings, you can ignore.